### PR TITLE
Add reverse streaming to FirebaseBigCollection.

### PR DESCRIPTION
**Pull request by mykmartin**
Not surfaced in the PEC or mirrored in InMemoryBigCollection yet.

### CODE REVIEW

File: runtime/ts/storage/firebase-storage.ts [hunk 1]
```diff
@@ -1342,10 +1367,14 @@ class FirebaseBigCollection extends FirebaseStorageProvider {
   // Returns a FirebaseCursor id for paginated reads of the current version of this BigCollection.
   // The id should be passed to cursorNext() to retrive the contained entities. The cursor itself
   // is held internally by this collection so we can discard it once the stream read has completed.
-  async stream(pageSize) {
+  //
+  // By default items are returned in order of original insertion into the collection (with the
+  // caveat that items removed during a streamed read may be returned at the end). Set forward to
+  // false to return items in reverse insertion order.
+  async stream(pageSize, {forward = true} = {}) {
```
Comment (piotrswigon) [inline runtime/ts/storage/firebase-storage.ts]: does it make sense to have both pageSize and forward as options set by the options object?

e.g. `async stream({pageSize = 10, forward = true} = {})`

File: runtime/ts/storage/firebase-storage.ts [hunk 2]
```diff
@@ -1160,17 +1162,21 @@ class FirebaseCursor {
...
-      if (snapshot.val().index <= this.end &&
-          (this.nextStart === null || snapshot.val().index >= this.nextStart)) {
+      const index = snapshot.val().index;
+      if (index > this.end) return;
+      if (this.nextBoundary === null || (this.forward && index >= this.nextBoundary)
+                                     || (!this.forward && index <= this.nextBoundary)) {
         this.removed.push(snapshot.val().value);
       }
-    });
+    });
```
Comment (piotrswigon) [inline runtime/ts/storage/firebase-storage.ts]: nit: maybe configure your editor to clear trailing spaces on save?

Comment (mykmartin) [reply runtime/ts/storage/firebase-storage.ts]: While I would prefer that, our code base has lots of trailing spaces in it.

File: runtime/ts/storage/firebase-storage.ts (hunk 1 above)
Comment (mykmartin) [reply runtime/ts/storage/firebase-storage.ts]: Yes, it probably should. I explicitly don't want to set a default for that, though.

Comment (mykmartin) [reply runtime/ts/storage/firebase-storage.ts]: I might change that in a separate PR; it affects the in-memory version as well (and I have a couple of more CLs ready for this feature).

**Pull request merged**